api/src/test/java/com/epam/pipeline/manager/docker/DockerRegistryNotificationTest.java

```diff
@@ -16,326 +16,394 @@
 
-        when(dockerClientFactoryMock.getDockerClient(any(DockerRegistry.class)))
-                .thenReturn(dockerClient);
-        when(dockerClient.getImageTags(any(String.class), any(String.class)))
-                .thenReturn(Collections.singletonList("latest"));
+        toolGroup1 = getToolGroup(ID, IMAGE_GROUP_1, ID, ANOTHER_SIMPLE_USER);
+        final ToolGroup toolGroup2 = getToolGroup(null, IMAGE_GROUP_2, ID, ANOTHER_SIMPLE_USER);
+        final ToolGroup toolGroup3 = getToolGroup(null, IMAGE_GROUP_1, ID_2, ANOTHER_SIMPLE_USER);
 
-        ToolVersion toolVersion = new ToolVersion();
-        toolVersion.setDigest("test_digest");
-        toolVersion.setSize(DOCKER_SIZE);
-        toolVersion.setVersion("test_version");
-        Mockito.doReturn(toolVersion).when(dockerClient)
-                .getVersionAttributes(Mockito.any(), Mockito.any(), Mockito.any());
+        tool1 = getTool(toolGroup1, IMAGE, registry, ANOTHER_SIMPLE_USER);
+        tool2 = getTool(toolGroup2, IMAGE_NEW_GROUP, registry, ANOTHER_SIMPLE_USER);
+        tool3 = getTool(toolGroup3, IMAGE, registryWithExternalPath, ANOTHER_SIMPLE_USER);
 
-        doNothing().when(toolVersionManager).updateOrCreateToolVersion(anyLong(), anyString(), anyString(),
-                any(DockerRegistry.class), any(DockerClient.class));
+        mockUserContext();
+        MockitoAnnotations.initMocks(this);
     }
 
     @Test
-    @Transactional(propagation = Propagation.REQUIRES_NEW, rollbackFor = Exception.class)
-    @WithMockUser(username = ADMIN, roles = ADMIN_ROLE)
-    public void testEventCanEnableToolInExistingGroupWithRightUserAccess() {
-        DockerRegistryEventEnvelope eventsEnvelope = generateDockerRegistryEvents(
-                Collections.singletonList(TEST_USER),
-                Collections.singletonList(TEST_REPO),
-                Collections.singletonList(TEST_IMAGE),
+    public void shouldEventEnableToolInExistingGroupWithRightUserAccess() {
+        mockRegistryDao(registry, REPO);
+        mockMetadataCheck(registry);
+        mockToolGroupExistence();
+        mockToolCreation(IMAGE, ID, tool1, ID);
+        initAclEntity(registry, AclPermission.WRITE);
+        initAclEntity(toolGroup1, AclPermission.WRITE);
+
+        final DockerRegistryEventEnvelope eventsEnvelope = generateDockerRegistryEvents(
+                Collections.singletonList(SIMPLE_USER),
+                Collections.singletonList(REPO),
+                Collections.singletonList(IMAGE),
                 Collections.singletonList(PUSH_ACTION));
 
-        List<Tool> registeredTools = registryManager.notifyDockerRegistryEvents(TEST_REPO, eventsEnvelope);
+        final List<Tool> registeredTools = registryManager.notifyDockerRegistryEvents(REPO, eventsEnvelope);
         Assert.assertEquals(1, registeredTools.size());
+
+        verify(mockDockerClientFactory).getDockerClient(eq(registry), eq(null));
```

Comment:
To my mind we don't need `verify` here, `assertEquals` would be enough.

Reply:
Done!